### PR TITLE
26 lunch dinner toggle

### PR DESCRIPTION
**Pull request by lchristus**

### CODE REVIEW

File: src/MensaWidget/MWFrame.class.st [hunk 1]
```diff
@@ -40,94 +41,105 @@ MWFrame class >> contentExtent [
...
-	#'squeak_changestamp' : 'Oscar 6/16/2024 23:52'
+	#'squeak_changestamp' : 'Louis Christ 6/18/2024 17:40'
 }
-MWFrame >> buildContentMorph [
+MWFrame >> addContentMorph [
 
-	^ AlignmentMorph newColumn 
+	self contentMorph: (AlignmentMorph newColumn 
 		 cellInset: self class cellSpacing;
 		 extent: self class contentExtent;
```
Comment (Kangaroo6942) [inline src/MensaWidget/MWFrame.class.st:80]: you didn't work on those lines, i just noticed that they're not correctly indented

File: src/MensaWidget/MWFrame.class.st [hunk 2]
```diff
@@ -40,94 +41,105 @@ MWFrame class >> contentExtent [
...
-		vResizing: #spaceFill;
-		extent: self class contentExtent;
-		color: MWColorTheme backgroundColor;
-		addMorph: self buildScrollPane;
-		addMorph: self buildMenuBar;
-		yourself
+	| morph |
+	
+	(MWDataGettingWrapper getMealsOnDate: aDate yyyymmdd) do: 
+		[:each | morph := (each buildForTime: aTime).
```
Comment (Kangaroo6942) [inline src/MensaWidget/MWFrame.class.st:95]: those brackets can be removed

File: src/MensaWidget/MWFrame.class.st [hunk 3]
```diff
@@ -40,94 +41,105 @@ MWFrame class >> contentExtent [
...
 ]
 
 {
-	#category : #creation,
-	#'squeak_changestamp' : 'Oscar 6/16/2024 23:51'
+	#category : #dayManipulation,
+	#'squeak_changestamp' : 'Louis Christ 6/18/2024 16:05'
 }
-MWFrame >> buildLayout [
+MWFrame >> addMeals: aDate forTime: aTime [
```
Comment (Kangaroo6942) [inline src/MensaWidget/MWFrame.class.st:90]: i think a name like `addMealsOn: aDate forTime: aTime` would be a bit more fitting

File: src/MensaWidget/MWFrame.class.st [hunk 4]
```diff
@@ -40,94 +41,105 @@ MWFrame class >> contentExtent [
...
-		color: MWColorTheme backgroundColor;
-		addMorph: self buildScrollPane;
-		addMorph: self buildMenuBar;
-		yourself
+	| morph |
+	
+	(MWDataGettingWrapper getMealsOnDate: aDate yyyymmdd) do: 
+		[:each | morph := (each buildForTime: aTime).
+		self contentMorph addMorph: morph.
+		morph vResizing: #shrinkWrap].
```
Comment (Kangaroo6942) [inline src/MensaWidget/MWFrame.class.st:97]: this dot can be removed

File: src/MensaWidget/MWFrame.class.st [hunk 5]
```diff
@@ -149,52 +161,80 @@ MWFrame >> contentMorph: anObject [
...
-MWFrame >> displayDay: aDate [
+MWFrame >> initialize [
 
-	self removeMeals;
-		addMeals: aDate
+	super initialize.
+	self listDirection: #topToBottom;
+		buildLayout;
+		addMeals: Date today forTime: MWMealCleaner lunchCategory;
+		openInWindowLabeled: self class windowName.
```
Comment (Kangaroo6942) [inline src/MensaWidget/MWFrame.class.st:173]: this dot can be removed

File: src/MensaWidget/MWFrameTest.class.st [hunk 6]
```diff
@@ -38,18 +38,11 @@ MWFrameTest >> tearDown [
...
 
-	self widgetFrame submorphsDo: [ :each | 
-		each class = ScrollPane ifTrue: 
-			[ each submorphsDo: [ :each2 |
-				each2 class = TransformationMorph ifTrue: 
-					[ each2 submorphsDo: [ :each3 |
-						each3 class = AlignmentMorph ifTrue: [
-							self assert: each3 subclasses size = 
-								MWDataGettingWrapper canteenDict size]]]]]]
+	self assert: self widgetFrame submorphs first contentMorph submorphs size = MWDataGettingWrapper canteenDict size
```
Comment (Kangaroo6942) [inline src/MensaWidget/MWFrameTest.class.st:45]: i think for better understanding of the testcase, something like widgetframe submorphs first could be put into a variable with descriptive name

File: src/MensaWidget/MWMenuBar.class.st [hunk 7]
```diff
@@ -224,12 +279,80 @@ MWMenuBar >> target: anObject [
...
+]
+
+{
+	#category : #timeManipulation,
+	#'squeak_changestamp' : 'Louis Christ 6/17/2024 21:51'
+}
+MWMenuBar >> toggleTime [
+
+	self time = 'lunch' ifTrue:
+		[self time: 'dinner'] ifFalse:
```
Comment (Kangaroo6942) [inline src/MensaWidget/MWMenuBar.class.st:324]: i think the ifTrue and the IfFalse should each move one line down, so it's immediately visible that we have two cases

File: src/MensaWidget/MWMenuBarTest.class.st [hunk 8]
```diff
@@ -66,95 +83,146 @@ MWMenuBarTest >> mensaWidget: anObject [
...
+
+	| oldTime |
+	
+	oldTime := self menuBar time.
+	self getAllCategories do: [:each | self assert: (each = oldTime or: [each = nil])].
+	self menuBar toggleTime.
+	self assert: (oldTime = self menuBar time) not.
+	self getAllCategories do: [:each | self assert: (each = oldTime) not].
+	self menuBar toggleTime.
+	self assert: oldTime = self menuBar time.
```
Comment (Kangaroo6942) [inline src/MensaWidget/MWMenuBarTest.class.st:154]: the dot is not needed

**Review by Kangaroo6942 — CHANGES_REQUESTED** [review]
Looks good, just the class comments should be adapted accordingly as well as the documentation

File: src/MensaWidget/MWColorTheme.class.st [hunk 9]
```diff
@@ -13,36 +13,36 @@ Class {
...
 {
 	#category : #colors,
-	#'squeak_changestamp' : 'Louis Christ 6/10/2024 17:41'
+	#'squeak_changestamp' : 'Louis Christ 6/9/2024 16:26'
 }
-MWColorTheme class >> transparentColor [
+MWColorTheme class >> secondaryColor [
 
-	^  Color transparent
+	^  Color fern lighter
```
Comment (OscarUlb) [inline src/MensaWidget/MWColorTheme.class.st:47]: There are two spaces before "Color"

File: src/MensaWidget/MWFrame.class.st [hunk 10]
```diff
@@ -40,94 +41,105 @@ MWFrame class >> contentExtent [
...
-			color: MWColorTheme backgroundColor;
-			extent: self class contentExtent;
+			color: MWColorTheme defaultColor;
 			hResizing: #spaceFill;
 			vResizing: #spaceFill;
-			yourself);
-		contentMorph
+			scroller);
+		addMorph: self scroller owner.
+	self scroller layoutPolicy: TableLayout new.
```
Comment (OscarUlb) [inline src/MensaWidget/MWFrame.class.st:126]: I think the line can be added to the cascade.

File: src/MensaWidget/MWMenuBar.class.st (hunk 7 above)
Comment (lchristus) [reply src/MensaWidget/MWMenuBar.class.st:324]: I like it as well we just need to align as a team how we format if statements

File: src/MensaWidget/MWFrame.class.st (hunk 10 above)
Comment (lchristus) [reply src/MensaWidget/MWFrame.class.st:126]: I dont think it can because there is more than one folloing method call

**Review by Kangaroo6942 — APPROVED** [review]
all resolved

**Review by OscarUlb — APPROVED** [review]

**Review by Kangaroo6942 — APPROVED** [review]

**Pull request merged**